Mathlib/Analysis/Convex/Normed.lean

```diff
@@ -153,3 +155,51 @@ theorem isConnected_setOf_sameRay_and_ne_zero {x : E} (hx : x ≠ 0) :
   simp_rw [← exists_pos_left_iff_sameRay_and_ne_zero hx]
   exact isConnected_Ioi.image _ (continuous_id.smul continuous_const).continuousOn
 #align is_connected_set_of_same_ray_and_ne_zero isConnected_setOf_sameRay_and_ne_zero
+
+end SeminormedAddCommGroup
+
+section NormedAddCommGroup
+variable [NormedAddCommGroup E] [NormedSpace ℝ E] [FiniteDimensional ℝ E] {s : Set E} {x : E}
+
+/-- We can intercalate a polyhedron between a point and one of its neighborhoods. -/
+lemma exists_mem_interior_convexHull_finset (hs : s ∈ 𝓝 x) :
+    ∃ t : Finset E, x ∈ interior (convexHull ℝ t : Set E) ∧ convexHull ℝ t ⊆ s := by
+  clear P ι
+  classical
+  wlog hx : x = 0
+  · obtain ⟨t, ht⟩ := this (s := -x +ᵥ s) (by simpa using vadd_mem_nhds (-x) hs) rfl
+    use x +ᵥ t
+    simpa [subset_set_vadd_iff, mem_vadd_set_iff_neg_vadd_mem, convexHull_vadd, interior_vadd]
+      using ht
+  subst hx
+  obtain ⟨b⟩ := exists_affineBasis_of_finiteDimensional
+    (ι := Fin (finrank ℝ E + 1)) (k := ℝ) (P := E) (by simp)
+  obtain ⟨ε, hε, hεs⟩ := Metric.mem_nhds_iff.1 hs
+  set u : Finset E := -Finset.univ.centroid ℝ b +ᵥ Finset.univ.image b
```

Comment:
I think a comment might be useful here and perhaps one other place in the proof

Reply:
Yes, that's exactly what's happening

Reply:
I've now added comments throughout